src/commands/installDocker.ts

```diff
@@ -5,10 +5,19 @@
 
 import * as os from 'os';
 import { IActionContext } from 'vscode-azureextensionui';
+import { localize } from '../localize';
+import { DockerExtensionKind, getVSCodeRemoteInfo } from '../utils/getVSCodeRemoteInfo';
 import { openExternal } from '../utils/openExternal';
 import { MacDockerInstaller, WindowsDockerInstaller } from './dockerInstaller';
 
 export async function installDocker(context: IActionContext): Promise<void> {
+    const remoteInfo = getVSCodeRemoteInfo(context);
```

Comment:
@ravipal I looked into `registerWorkspaceCommand`, it's essentially made to warn you if you are in UI mode but try to do something like `docker build`, which must always run in the workspace. So, it doesn't do what we need.

Instead I put this block in; if remoting is in use then we definitely don't want to get into the business of trying to install Docker (plus the target of remoting would almost always be Linux where we don't have support anyway).

Can you take a look at this code and see if it looks OK?

Reply:
Can this check be added during register command? something like `registerLocalCommand` so in future other such commands can reuse the code.

Reply:
Good idea! I'll do that.